crates/forge/src/runner.rs

```diff
@@ -650,4 +668,51 @@ impl<'a> ContractRunner<'a> {
         }
         test_result.fuzz_result(result)
     }
+
+    /// Prepares single unit test and fuzz test execution:
+    /// - set up the test result and executor
+    /// - check if before test txes are configured and apply them in order
+    ///
+    /// Before test txes are arrays of arbitrary calldata obtained by calling the `beforeTest`
+    /// function with test selector as a parameter.
+    ///
+    /// Unit tests within same contract (or even current test) are valid options for before test tx
+    /// configuration. Test execution stops if any of before test txes fails.
+    fn prepare_test(
+        &self,
+        func: &Function,
+        setup: TestSetup,
+    ) -> Result<(Executor, TestResult, Address), TestResult> {
+        let address = setup.address;
+        let mut executor = Cow::Borrowed(&self.executor);
+        let mut test_result = TestResult::new(setup);
+
+        // Apply before test configured functions (if any).
+        if self.contract.abi.functions().filter(|func| func.name.is_before_test_setup()).count() ==
+            1
+        {
+            for calldata in executor
+                .call_sol_default(
+                    address,
+                    &ITest::beforeTestSetupCall { testSelector: func.selector() },
+                )
+                .beforeTestCalldata
+            {
+                // Apply before test configured calldata.
+                match executor.to_mut().transact_raw(self.sender, address, calldata, U256::ZERO) {
+                    Ok(call_result) => {
+                        // Merge tx result traces in unit test result.
+                        test_result.merge_call_result(&call_result);
+
+                        // To continue unit test execution the call should not revert.
+                        if call_result.reverted {
+                            return Err(test_result.single_fail(None))
+                        }
+                    }
+                    Err(_) => return Err(test_result.single_fail(None)),
+                }
+            }
+        }
+        Ok((executor.into_owned(), test_result, address))
```

Comment:
the Cow should be returned for this to have any effect. into_owned clones if Borrowed

Reply:
sorry, got this wrong :man_facepalming:  pls check https://github.com/foundry-rs/foundry/pull/8497/commits/2360fb1ec9ac00536b04312dfcb6ffc89835db90